lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -585,6 +586,14 @@ bool canFoldIntoConversion(Operation *op, Attribute targetEncoding) {
            !triton::gpu::isExpensiveView(reshape.getSrc().getType(),
                                          newDstType);
   }
+  if (auto tmemLoad = dyn_cast<ttn::TMEMLoadOp>(op)) {
+    auto mod = op->getParentOfType<ModuleOp>();
+    auto memDescType = tmemLoad.getSrc().getType();
+    auto oldRetType = tmemLoad.getResult().getType();
+    auto type = RankedTensorType::get(
+        oldRetType.getShape(), oldRetType.getElementType(), targetEncoding);
+    return ttn::isDistributedLayoutTMemCompatible(mod, type, memDescType);
```

Comment:
technically correct, however I don't think `isDistributedLayoutTMemCompatible` is full proof and I believe there is only one combination of distributed and tmem layout we actually support  in the lowering of TMemLoad

Reply:
Oh well, that's sad. Let me close this PR  until we have more generic support for tmemloads